src/openzaak/utils/tasks.py

```diff
@@ -0,0 +1,791 @@
+
+    gegevensgroep_fields = ("ondertekening", "integriteit")
+
+    for field in gegevensgroep_fields:
+        gegevens_groep_value = data.pop(field, {})
+
+        for key, value in gegevens_groep_value.items():
+            data[f"{field}_{key}"] = value
+
+    if "vertrouwelijkheidaanduiding" not in data:
+        informatieobjecttype = data["informatieobjecttype"]
+        data[
+            "vertrouwelijkheidaanduiding"
+        ] = informatieobjecttype.vertrouwelijkheidaanduiding
+
+    instance = EnkelvoudigInformatieObject(**data)
+
+    if not instance.identificatie:
+        instance.identificatie = generate_unique_identification(
+            instance, "creatiedatum"
+        )
+
+    try:
+        instance.clean()
+    except ValidationError as e:
+        error_message = (
+            "A validation error occurred while validating a "
+            f"EnkelvoudigInformtatieObject on line {row_index}: \n"
+            f"{str(e)}"
+        )
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    file_path = document_row.bestandspad
+    path = Path(settings.IMPORT_DOCUMENTEN_BASE_DIR) / Path(file_path)
+
+    if not path.exists() or not path.is_file():
+        error_message = (
+            f"The given filepath {path} does not exist or is a file for "
+            f"row {row_index}"
+        )
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    default_dir = _get_default_path(instance.inhoud)
+    import_path = f"{default_dir}/{path.name}"
+
+    if not default_dir.exists():
+        default_dir.mkdir(parents=True)
+
+    try:
+        shutil.copy2(path, import_path)
+    except Exception as e:
+        error_message = f"Unable to copy file for row {row_index}: \n {str(e)}"
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    instance.inhoud.name = str(import_path)
+
+    document_row.instance = instance
+    return document_row
+
+
+@transaction.atomic()
+def _batch_create_eios(batch: list[DocumentRow], zaak_uuids: dict[str, int]) -> None:
+    canonicals = []
+
+    for row in batch:
+        if row.instance is None:
+            continue
+
+        canonical = EnkelvoudigInformatieObjectCanonical()
+        row.instance.canonical = canonical
+
+        canonicals.append(canonical)
+
+    try:
+        EnkelvoudigInformatieObjectCanonical.objects.bulk_create(canonicals)
+    except IntegrityError as e:
+        for row in batch:
+            row.processed = True
+            row.comment = f"Unable to load row due to batch error: {str(e)}"
+
+        raise e
+
+    try:
+        eios = EnkelvoudigInformatieObject.objects.bulk_create(
+            [row.instance for row in batch if row.instance is not None]
```

Comment:
Does this mean that not all rows have to succeed for the import to have effect? Is this intentional?

Reply:
Before feeding all the instances to the `bulk_create`, various validation checks are done during `_import_document_row`. Because the creation of all the instances in the function run in a single transaction, all of these instances would fail whenever one of them would have a validation error and no validation was done beforehand. Hence the filtering being doing here (e.g no instance on the row means it did not pass validation) and the validation beforehand in `_import_document_row`.